src/librustc_target/spec/wasm32_unknown_emscripten.rs

```diff
@@ -1,45 +1,43 @@
-use super::{LinkArgs, LinkerFlavor, Target, TargetOptions};
+use super::wasm32_base;
+use super::{LinkArgs, LinkerFlavor, Target, TargetOptions, PanicStrategy};
 
 pub fn target() -> Result<Target, String> {
-    // FIXME(nikic) BINARYEN_TRAP_MODE=clamp is needed to avoid trapping in our
-    // -Zsaturating-float-casts implementation. This can be dropped if/when
-    // we have native fpto[su]i.sat intrinsics, or the implementation otherwise
-    // stops relying on non-trapping fpto[su]i.
     let mut post_link_args = LinkArgs::new();
     post_link_args.insert(LinkerFlavor::Em,
                           vec!["-s".to_string(),
-                               "BINARYEN=1".to_string(),
-                               "-s".to_string(),
                                "ERROR_ON_UNDEFINED_SYMBOLS=1".to_string(),
                                "-s".to_string(),
-                               "BINARYEN_TRAP_MODE='clamp'".to_string()]);
+                               "ASSERTIONS=1".to_string(),
+                               "-s".to_string(),
+                               "DISABLE_EXCEPTION_CATCHING=0".to_string(),
+                               // FIXME(tlively): Enable this linker option once libc type errors
+                               // are resolved. See https://github.com/rust-lang/libc/pull/1478.
+                               // "-Wl,--fatal-warnings".to_string(),
+                               ]);
 
     let opts = TargetOptions {
-        dynamic_linking: false,
-        executables: true,
-        // Today emcc emits two files - a .js file to bootstrap and
-        // possibly interpret the wasm, and a .wasm file
+        // emcc emits two files - a .js file to instantiate the wasm and supply platform
+        // functionality, and a .wasm file.
         exe_suffix: ".js".to_string(),
+        linker: None,
         linker_is_gnu: true,
-        allow_asm: false,
-        obj_is_bitcode: true,
         is_like_emscripten: true,
-        max_atomic_width: Some(32),
+        // FIXME(tlively): Emscripten supports unwinding, but we would have to pass
```

Comment:
@alexcrichton I couldn't figure out how to pass LLVM codegen arguments (although I did see how to pass linker arguments). Do you know how?

Reply:
Oh oops sorry missed this! I don't think we currently have an option for LLVM arguments, but feel free to add an option and I think it'd end up being read [around here](https://github.com/rust-lang/rust/blob/f86521e0a33a2b54c4c23dbfc5250013f7a33b11/src/librustc_codegen_llvm/llvm_util.rs#L46-L83)